test/tools/fuzzer_common.cpp

```diff
@@ -17,28 +17,87 @@
 
 #include <test/tools/fuzzer_common.h>
 
+#include <libsolidity/interface/CompilerStack.h>
+
 #include <libsolutil/JSON.h>
+
 #include <libevmasm/Assembly.h>
 #include <libevmasm/ConstantOptimiser.h>
+
 #include <libsolc/libsolc.h>
 
+#include <liblangutil/Exceptions.h>
+
 #include <sstream>
 
 using namespace std;
 using namespace solidity;
 using namespace solidity::util;
 using namespace solidity::evmasm;
-
-static vector<string> s_evmVersions = {
-	"homestead",
-	"tangerineWhistle",
-	"spuriousDragon",
-	"byzantium",
-	"constantinople",
-	"petersburg",
-	"istanbul"
+using namespace solidity::langutil;
+
+static vector<EVMVersion> s_evmVersions = {
+	EVMVersion::homestead(),
+	EVMVersion::tangerineWhistle(),
+	EVMVersion::spuriousDragon(),
+	EVMVersion::byzantium(),
+	EVMVersion::constantinople(),
+	EVMVersion::petersburg(),
+	EVMVersion::istanbul(),
+	EVMVersion::berlin()
 };
 
+void FuzzerUtil::testCompilerJsonInterface(string const& _input, bool _optimize, bool _quiet)
+{
+	if (!_quiet)
+		cout << "Testing compiler " << (_optimize ? "with" : "without") << " optimizer." << endl;
+
+	Json::Value config = Json::objectValue;
+	config["language"] = "Solidity";
+	config["sources"] = Json::objectValue;
+	config["sources"][""] = Json::objectValue;
+	config["sources"][""]["content"] = _input;
+	config["settings"] = Json::objectValue;
+	config["settings"]["optimizer"] = Json::objectValue;
+	config["settings"]["optimizer"]["enabled"] = _optimize;
+	config["settings"]["optimizer"]["runs"] = 200;
+	config["settings"]["evmVersion"] = "berlin";
+
+	// Enable all SourceUnit-level outputs.
+	config["settings"]["outputSelection"]["*"][""][0] = "*";
+	// Enable all Contract-level outputs.
+	config["settings"]["outputSelection"]["*"]["*"][0] = "*";
+
+	runCompiler(jsonCompactPrint(config), _quiet);
+}
+
+void FuzzerUtil::testCompiler(string const& _input, bool _optimize)
+{
+	frontend::CompilerStack compiler;
+	EVMVersion evmVersion = s_evmVersions[_input.size() % s_evmVersions.size()];
+	frontend::OptimiserSettings optimiserSettings;
+	if (_optimize)
+		optimiserSettings = frontend::OptimiserSettings::standard();
+	else
+		optimiserSettings = frontend::OptimiserSettings::none();
```

Comment:
Why not minimal?

Reply:
Ah, I didn't notice it earlier.

https://github.com/ethereum/solidity/blob/d55bbd4aa5dd24ca6f105b24d87127679e1f1810/libsolidity/interface/StandardCompiler.cpp#L454